modules/core/shared/src/main/scala/lucuma/core/util/TimestampInterval.scala

```diff
@@ -0,0 +1,236 @@
+// Copyright (c) 2016-2023 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.core.util
+
+import cats.Comparison
+import cats.Eq
+import cats.Order
+import cats.Order.catsKernelOrderingForOrder
+import cats.syntax.order.*
+import org.typelevel.cats.time.*
+
+import java.time.Instant
+
+/**
+ * A TimestampInterval represents a period of time with a fixed starting point
+ * 'start' (inclusive) and an fixed ending point 'end' (exclusive).  In other
+ * words, [start, end).  Whereas a TimeSpan focuses solely on an amount of
+ * time, the TimestampInterval tracks specific start and end times.
+ *
+ * @param start time at which the interval begins (inclusive)
+ * @param end time at which the interval ends (exclusive)
+ */
+sealed class TimestampInterval private (val start: Timestamp, val end: Timestamp) {
```

Comment:
Nice.

My first impression was that this could `extend BoundedInterval[Timestamp]` but upon closer inspection not all methods convey the same information, especially information about left-closed, right-open resulting intervals.

But there are number of methods that seems the same: `abuts`, `contains`, `intersects` at least. 

Maybe this class can wrap a `BoundedInterval[Timestamp]` and just export the methods that are the same? Maybe it can rely on other logic from `BoundedInterval` for other operations too.